src/main/java/de/tum/in/www1/artemis/service/connectors/bitbucket/BambooBuildPlanUpdateService.java

```diff
@@ -0,0 +1,184 @@
+     * second using the secondRepositoryName
+     *
+     * @param firstRepositoryName  the repository name that should be found first
+     * @param secondRepositoryName in case firstRepositoryName is not found, the repository name that should be found second
+     * @param buildPlanKey         the bamboo build plan key in which the repository with the specified name should be found
+     * @return the found bamboo repository in the build plan or null in case the repository with the specified name could not be found
+     */
+    private BambooRepositoryDTO findBambooRepository(String firstRepositoryName, @Nullable String secondRepositoryName, String buildPlanKey) {
+        List<BambooRepositoryDTO> list = this.getBuildPlanRepositoryList(buildPlanKey);
+        var bambooRepository = this.lookupRepository(firstRepositoryName, list);
+        if (bambooRepository == null && secondRepositoryName != null) {
+            bambooRepository = this.lookupRepository(secondRepositoryName, list);
+        }
+        return bambooRepository;
+    }
+
+    private BambooRepositoryDTO lookupRepository(String name, List<BambooRepositoryDTO> list) {
+        var repository = list.stream().filter(repo -> name.equals(repo.getName())).findFirst();
+        if (repository.isPresent()) {
+            return repository.get();
+        }
+
+        if (StringUtils.isNumeric(name)) {
+            Long repoId = Long.valueOf(name);
+            repository = list.stream().filter(repo -> repoId.equals(repo.getId())).findFirst();
+            if (repository.isPresent()) {
+                return repository.get();
+            }
+        }
+
+        return null;
+    }
+
+    private List<BambooRepositoryDTO> getBuildPlanRepositoryList(String buildPlanKey) {
+        List<BambooRepositoryDTO> list = new ArrayList<>();
+        String requestUrl = bambooServerUrl + "/chain/admin/config/editChainRepository.action";
+        UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(requestUrl).queryParam("buildKey", buildPlanKey);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setAccept(Collections.singletonList(MediaType.TEXT_HTML));
+        HttpEntity<String> entity = new HttpEntity<>(null, headers);
+        String data = bambooRestTemplate.exchange(builder.build().toUri(), HttpMethod.GET, entity, String.class).getBody();
+
+        // the following code finds the required information in the html response
+        // it iterates over all data-item-ids in the html code and tries to find the required repository information inside
+        // please note: while this approach is not ideal, it works reliably, and there is currently no other API to get the repository id which is needed for other calls to Bamboo
+        if (data != null) {
+            Pattern findPattern = Pattern.compile("data-item-id=\"(\\d+)\".*\"item-title\">([^<]+)<", Pattern.DOTALL);
```

Comment:
This is just a standard Regex to detect HTML attributes, right? If so, can you add a comment explaining this?